src/Autocomplete/src/Form/WrappedEntityTypeAutocompleter.php

```diff
@@ -99,7 +99,17 @@ public function getLabel(object $entity): string
 
     public function getValue(object $entity): string
     {
-        return $this->getEntityMetadata()->getIdValue($entity);
+        $choiceValue = $this->getFormOption('choice_value');
+
+        if (\is_string($choiceValue) || $choiceValue instanceof PropertyPathInterface) {
+            return $this->propertyAccessor->getValue($entity, $choiceValue);
```

Comment:
The return type of the method is not compatible with all the new possibilities no ? Integers, floats, nulls, ...

Reply:
Historically, I think there's already a problem when we see the return type of `getIdentifierValues()`:

```
/**
     * Returns the identifier of this object as an array with field name as key.
     *
     * Has to return an empty array if no identifier isset.
     *
     * @return array<string, mixed>
     */
    public function getIdentifierValues(object $object);
```

But there is no problem because is always cast to string.

> By default, PHP will coerce values of the wrong type into the expected scalar type declaration if possible. For example, a function that is given an int for a parameter that expects a string will get a variable of type string.